### PR TITLE
Feature/disable collection count

### LINKED ISSUE #3: Separate method in the alchemy layer for collection count fetching
Sometimes we just don't need the collection count for the model/resource or the query and/or the table is so huge, that it would take a lot of time to fetch the collection count.
It happens inside the `get_collection` here: https://github.com/AdCombo/flask-rest-jsonapi/blob/725cf11286a1ea06aa2ca62182a212d03fceb1c8/flask_rest_jsonapi/data_layers/alchemy.py#L160

So, the decision is:
- create a separate method in the `SqlalchemyDataLayer` for getting the collection count. For example `get_collection_count`. So it can be easily overridden and being replaced with anything else 
- create `disable_collection_count` and `default_collection_count`, get it from the resource

### PR DESCRIPTION
**Pull request by mahenzon**
Closes #3 

- Create a separate method get_collection_count in the alchemy layer https://github.com/AdCombo/flask-rest-jsonapi/pull/4/commits/39c1498f24346ed391f1316f252e9f49fef85541
- Create post_init method for base data layer, use it in the resource https://github.com/AdCombo/flask-rest-jsonapi/pull/4/commits/e51c676c603d09577c8a39b85e34a9ce6aa38cd9. This method is called after the resource is assigned to the layer. Also make the `resource` attribute on the data layer mandatory, though optional
- Override `post_init` and set `disable_collection_count` and `default_collection_count` in alchemy layer https://github.com/AdCombo/flask-rest-jsonapi/pull/4/commits/46df5de0d13c814aff84f02405dacc88837239cc
- tests coverage https://github.com/AdCombo/flask-rest-jsonapi/pull/4/commits/0e7731074be488093b8798512873658ad45e3d0a

### CODE REVIEW

**Comment by mahenzon** [comment]
This PR is now blocked by #5 and needs a rebase after merging that one. I extracted refactoring, so this PR is easier to review

**Review by Znbiz — COMMENTED** [review]
All is well, you can merge

**Review by Znbiz — COMMENTED** [review]
All is well, you can merge

**Review by Znbiz — APPROVED** [review]

**Pull request merged**
